File: backend/src/test/java/ca/metricalsky/winston/entity/VideoCommentsEntityTest.java
package ca.metricalsky.winston.entity;

import ca.metricalsky.winston.test.TestUtils;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;

import static org.assertj.core.api.Assertions.assertThat;

@DataJpaTest
class VideoCommentsEntityTest {

    @Autowired
    private TestEntityManager entityManager;

    private VideoEntity videoEntity;

    @BeforeEach
    void beforeEach() {
        var channelEntity = persistChannel();
        videoEntity = persistVideo(channelEntity);
    }

    @Test
    void persists() {
        var videoCommentsEntity = VideoCommentsEntity.builder()
                .videoId(videoEntity.getId())
                .commentsDisabled(TestUtils.randomBoolean())
                .commentCount(TestUtils.randomLong())
                .replyCount(TestUtils.randomLong())
                .totalReplyCount(TestUtils.randomLong())
                .build();

        var persistedEntity = entityManager.persistFlushFind(videoCommentsEntity);

        assertThat(persistedEntity)
                .hasFieldOrPropertyWithValue("videoId", videoCommentsEntity.getVideoId())
                .hasFieldOrPropertyWithValue("commentsDisabled", videoCommentsEntity.isCommentsDisabled())
                .hasFieldOrPropertyWithValue("commentCount", videoCommentsEntity.getCommentCount())
                .hasFieldOrPropertyWithValue("replyCount", videoCommentsEntity.getReplyCount())
                .hasFieldOrPropertyWithValue("totalReplyCount", videoCommentsEntity.getTotalReplyCount())
                .hasNoNullFieldsOrProperties();
    }


    private ChannelEntity persistChannel() {
        var channelEntity = ChannelEntity.builder()
                .id(TestUtils.randomId())
                .build();
        return entityManager.persist(channelEntity);
    }

    private VideoEntity persistVideo(ChannelEntity channelEntity) {
        var videoEntity = VideoEntity.builder()
                .id(TestUtils.randomId())
                .channelId(channelEntity.getId())
                .build();
        return entityManager.persist(videoEntity);
    }
}

File: database/src/test/kotlin/ca/metricalsky/winston/database/entity/video/VideoCommentsEntityTest.kt
package ca.metricalsky.winston.database.entity.video

import ca.metricalsky.winston.database.test.annotation.DatabaseTest
import ca.metricalsky.winston.database.test.faker.DatabaseFaker
import io.kotest.core.spec.style.WordSpec
import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.shouldBe
import io.kotest.matchers.shouldNotBe
import io.kotest.matchers.types.shouldNotBeSameInstanceAs
import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager

@DatabaseTest
class VideoCommentsEntityTest(
    entityManager: TestEntityManager,
): WordSpec({

    val faker = DatabaseFaker()

    "VideoCommentsEntity" should {

        "persist" {
            val channel = entityManager.persist(faker.channel().minimalEntity())
            val video = entityManager.persist(faker.video().minimalEntity(channel))
            val videoComments = faker.videoComments().completeEntity(video)

            entityManager.persistFlushFind(videoComments) shouldNotBeNull {
                this shouldNotBeSameInstanceAs videoComments
                videoId shouldBe videoComments.videoId
                commentsDisabled shouldBe videoComments.commentsDisabled
                commentCount shouldBe videoComments.commentCount
                replyCount shouldBe videoComments.replyCount
                totalReplyCount shouldBe videoComments.totalReplyCount
                lastFetchedAt shouldNotBe null
            }
        }
    }
})
